configure.d/1_queue_limits.conf

```diff
@@ -12,20 +12,25 @@ check() {
 	cur_name=$(basename $2)
 	config_file_path=$1
 
-	if compile_module $cur_name "struct queue_limits q; q.limits_aux;" "linux/blkdev.h"
+	if compile_module $cur_name "struct queue_limits q; q.max_write_zeroes_sectors;" "linux/blkdev.h"
 	then
-		echo $cur_name "1" >> $config_file_path
-	elif compile_module $cur_name "struct queue_limits q; q.max_write_zeroes_sectors;" "linux/blkdev.h"
-	then
-		if compile_module $cur_name "struct queue_limits q; q.max_write_same_sectors;" "linux/blkdev.h"
+		if compile_module $cur_name "struct queue_limits q; q.misaligned;" "linux/blkdev.h"
 		then
-			echo $cur_name "2" >> $config_file_path
+			if compile_module $cur_name "struct queue_limits q; q.max_write_same_sectors;" "linux/blkdev.h"
+			then
+				echo $cur_name "1" >> $config_file_path
+			else
+				echo $cur_name "2" >> $config_file_path
+			fi
 		else
 			echo $cur_name "3" >> $config_file_path
 		fi
 	elif compile_module $cur_name "struct queue_limits q; q.max_write_same_sectors;" "linux/blkdev.h"
 	then
 		echo $cur_name "4" >> $config_file_path
+	elif compile_module $cur_name "struct queue_limits q; q.limits_aux;" "linux/blkdev.h"
+	then
+		echo $cur_name "5" >> $config_file_path
 	else
 		echo $cur_name "X" >> $config_file_path
 	fi
```

Comment:
How about extracting `cas_queue_limits_is_misaligned()` to a separate file? The detection phase had already been quite complex and adding one more nested if makes it difficult to read. Another reason is that the new function is the only setter in this file. 
This change could be a separate commit